h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/strategy/ThresholdTEApplicationStrategy.java

```diff
@@ -0,0 +1,40 @@
+package ai.h2o.targetencoding.strategy;
+
+import water.fvec.Frame;
+import water.fvec.Vec;
+
+import java.util.ArrayList;
+
+/**
+ * Strategy that will select only categorical columns which cardinality is greater of equal than specified threshold
+ */
+public class ThresholdTEApplicationStrategy extends TEApplicationStrategy {
+  
+  private Frame _frame;
+  private String _responseColumnName;
+  private long _threshold;
+
+  /**
+   * 
+   * @param frame
+   * @param responseColumn
+   * @param threshold
+   */
+  public ThresholdTEApplicationStrategy(Frame frame, Vec responseColumn, long threshold) {
+    _frame = frame; 
+    _responseColumnName = _frame.name(_frame.find(responseColumn)); // Question: can we add .name(Vec vec) to Frame's API?
```

Comment:
I'd be in favour of answering "yes" to your question.

Reply:
`.name(Vec vec)`? I don't know what that would be good for

this is imho recipe for disaster:

     _frame.name(_frame.find(responseColumn))

you shouldn't be using the Vec to reverse look-up its name - for many reasons, including the very strange one that the Frame actually _can_ have 2 different columns backed by the same Vec

I would say this constructor should instead look like this:

    public ThresholdTEApplicationStrategy(Frame frame, String responseColumn, long threshold)

then there is no issue with reverse look-up and the API is impossible to use the wrong way

Reply:
(I also have the same objections toward `.find(Vec vec)` - which is an existing method - only used in MRUtils (btw) - I don't think that one is a good idea in general either

Reply:
but you have the same problem with `name` or `key` then, don't we?
`find(Vec)` is doing a lookup by Vec key, and so is `find(Key)`.
Btw, do we actually prevent duplicate Vec names in a frame? I'm not 100% sure it's the case, at least according to `Frame` constructor.

Reply:
Ok, convinced now. 
I would suggest to deprecate those public methods that are not recommended then, if they can lead to subtle bugs.
Also, regarding the signature, and given the usage mane of response column, it looks like it could be profitably replaced by `String[] excludedColumnNames` as not only response, but also `foldColumn` will need to get the same logic applied.
Same for `AllCategoricalTEApplicationStrategy` constructor.

Reply:
@michalkurka 
> you shouldn't be using the Vec to reverse look-up its name - for many reasons, including the very strange one that the Frame actually can have 2 different columns backed by the same Vec

I was relying on existing API. If we don't want to add `.name(Vec)` for mentioned above reasons then we should get rid of(deprecate) `.find(Vec)` and`.find(Key)`. So I agree with @seb-h2o regarding deprecation.

Regarding reverse look-up.... My first thought was that Vec should maybe know its name. But ok, design is different: Frame is a holder of vecs and maps names to vecs. This gives us a freedom to have multiple columns being backed by the same Vec - ok.  But if we are ok with  this design... I think it is eligible in general to ask Frame about "what is the name for a given Vec you have". So I would say reverse lookup on its own is a fine idea. But we allow columns to share Vecs and that is what makes this lookup dangerous. 

> I would say this constructor should instead look like this:

I would do it this way of course but the problem is that in AutoML we store responseColumn as a Vec (L224) and I would have to do this reverse lookup anyway either outside of the `TEApplicationStrategy` or inside. And as there were no other usages for `responseColumnName` as a String outside of the scope of `TEApplicationStrategy` I decided to move this reverse lookup logic inside the component increase cohesion.

Reply:
@seb-h2o Yeap. That would be enough! Thanks!

Reply:
I would say, in `AutoML`, feel free to change the package protected attributes:
```
Vec _responseColumn -> String _responseColumnName
```
and modify the public corresponding accessors accordingly:
```
  public Vec getResponseColumn() { return _trainingFrame.vec(_responseColumnName); }

```